bittide-instances/src/Bittide/Instances/Tests/VexRiscv.hs

```diff
@@ -0,0 +1,141 @@
+-- SPDX-FileCopyrightText: 2022-2023 Google LLC
+--
+-- SPDX-License-Identifier: Apache-2.0
+{-# LANGUAGE RecordWildCards #-}
+{-# OPTIONS_GHC -fplugin=Protocols.Plugin #-}
+
+module Bittide.Instances.Tests.VexRiscv where
+
+import Clash.Annotations.TH (makeTopEntity)
+import Clash.Cores.Xilinx.VIO (vioProbe)
+
+import Clash.Prelude
+
+import Bittide.Instances.Domains (Basic200, Basic125)
+import Bittide.DoubleBufferedRam
+import Language.Haskell.TH (runIO)
+import System.Directory
+import System.FilePath
+import Bittide.ProcessingElement
+import Bittide.ProcessingElement.Util (memBlobsFromElf)
+import Bittide.SharedTypes
+
+import Protocols
+import Protocols.Internal
+import Protocols.Wishbone
+import Clash.Xilinx.ClockGen (clockWizardDifferential)
+import Clash.Reset.Extra (noReset, orReset)
+import Clash.Cores.Xilinx.Xpm.Cdc.Single (xpmCdcSingle)
+
+data TestStatus = Running | Success | Fail
+  deriving (Enum, Eq, Generic, NFDataX, BitPack)
+
+vexRiscvInner ::
+  forall dom.
+  HiddenClockResetEnable dom =>
+  Signal dom (Bool, Bool)
+vexRiscvInner = stateToDoneSuccess <$> status
+  where
+
+    stateToDoneSuccess Running = (False, False)
+    stateToDoneSuccess Success = (True, True)
+    stateToDoneSuccess Fail    = (True, False)
+
+    unitC = CSignal (pure ())
+    (_, CSignal status) = circuitFn ((), unitC)
+
+    Circuit circuitFn = circuit $ \unit -> do
+        [wb] <- processingElement peConfig -< unit
+        statusRegister -< wb
+
+    statusRegister :: Circuit (Wishbone dom 'Standard 30 (Bytes 4)) (CSignal dom TestStatus)
+    statusRegister = Circuit $ \(fwd, CSignal _) ->
```

Comment:
@lmbollen Is there a generic component for this?

Reply:
There is `wbToVec` https://github.com/bittide/bittide-hardware/blob/b244b2af2de71ee36b26391670a820f3188a1482/bittide/src/Bittide/Wishbone.hs#L336 in #354 which could be adapted to be used in a circuit like this, but there's no generalised version like that AFAIK

Reply:
cool cool, let's leave it like it is

Reply:
We could definitely make a circuit version for `wbToVec`. The main reason I implemented it like this is to easily create interfaces with multiple signals. The reason I have not implemented it as `Signal` is to make it obvious that it's a simple combinatorial component.